File: src/api/supabase/actions/history.ts
import { supabase } from '../supabase';

// Añadir item al historial
export const addToHistory = async (userId: string, itemId: string) => {
  const { data, error } = await supabase
    .from('history')
    .insert({ user_id: userId, item_id: itemId, viewed_at: new Date() });

  if (error) throw new Error(error.message);
  return data;
};

// Obtener historial de un usuario
export const getHistory = async (userId: string) => {
  const { data, error } = await supabase
    .from('history')
    .select('*')
    .eq('user_id', userId)
    .order('viewed_at', { ascending: false });

  if (error) throw new Error(error.message);
  return data;
};

File: backend/routers/history.py
from fastapi import APIRouter, HTTPException
from supabase import create_client, Client
import os
import logging

router = APIRouter()
logger = logging.getLogger(__name__)

supabase: Client = create_client(
    os.getenv("SUPABASE_URL"),
    os.getenv("SUPABASE_SERVICE_ROLE_KEY")
)

@router.get("/{user_id}")
async def get_user_history(user_id: str):
    """
    Get user's recipe history.
    """
    try:
        resp = supabase.table("history") \
            .select("*, recipes(*)") \
            .eq("user_id", user_id) \
            .order("created_at", desc=True) \
            .execute()
        
        recipes = []
        for item in resp.data:
            if item.get("recipes"):
                recipes.append(item["recipes"])
        
        return recipes
    except Exception:
        logger.exception("Error fetching history")
        raise HTTPException(status_code=500, detail="Unable to fetch history at this time")